airflow/providers/amazon/aws/triggers/rds.py

```diff
@@ -129,16 +130,21 @@ def __init__(
         waiter_max_attempts: int,
         aws_conn_id: str,
         response: dict[str, Any],
-        db_type: RdsDbType,
+        db_type: RdsDbType | str,
         region_name: str | None = None,
     ) -> None:
+        # allow passing enums for users,
+        # but we can only rely on strings because (de-)serialization doesn't support enums
+        if isinstance(db_type, RdsDbType):
+            db_type = db_type.value
```

Comment:
Mypy is likely complaining this particular variable is a union type instead. Do this:

```python
if isinstance(db_type, RdsDbType):
    db_type_val = db_type.value
else:
    db_type_val = db_type
```

and change all `db_type` occurences to `db_type_val` and Mypy should be satisfied.